Table.cs

```diff
@@ -0,0 +1,187 @@
+using System.IO;
+using System.Collections.Generic;
+
+namespace Card_Game
+{
+    public class Table
+    {
+        private List<Player> players;
+        private TableDeck tableDeck;
+        private TableDeck roundDeck;
+        private TableDeck benchDeck;
+        private Player roundWinner;
+        private Dictionary<Card, Player> cardOwners;
+
+
+        public Table(params string[] playersNames) // first inserted player will start the game
+        {
+            CreateTableDeck();
+            CreateRoundDeck();
+            CreateBenchDeck();
+            CreatePlayers(playersNames);
+            CreateCardOwners();
+        }
+
+        private void CreateTableDeck()
+        {
+            string fileName = Path.GetFileName(Directory.GetFiles("files")[0]); // DAO selected according to the only file in the "files" directory
+            ICardsDAO fullDeck = GetProperDAO(fileName);
+            tableDeck = new TableDeck(fullDeck);
+        }
+
+        private ICardsDAO GetProperDAO(string fileName)
+        {
+            string extension = Path.GetExtension(fileName);
+            switch (extension)
+            {
+                case "csv":
+                    return new CardsDAO();
+                default:
+                    throw new FileNotFoundException("Not possible to upload cards.");
+            }
+        }
+
+        private void CreateRoundDeck()
+        {
+            roundDeck = new TableDeck();
+        }
+
+        private void CreateBenchDeck()
+        {
+            benchDeck = new TableDeck();
+        }
+
```

Comment:
Do we really need these two methods? We may initialize these decks in constructor or when creating fields

Reply:
Nope, already moved initialization to fields creation. Thanks!